substrafl/strategies/scaffold.py

```diff
@@ -39,7 +39,7 @@ def __init__(self, algo: Algo, aggregation_lr: float = 1):
             aggregation_lr (float, Optional): Global aggregation rate applied on the averaged weight updates
                 (`eta_g` in the paper). Defaults to 1. Must be >=0.
         """
-        super(Scaffold, self).__init__(algo=algo)
+        super(Scaffold, self).__init__(algo=algo, aggregation_lr=aggregation_lr)
```

Comment:
Isn't it a case where we want to use `*args, **kwargs` to pass all relevant parameters to the parent Strategy, without having to remember which are useful and which are not?

Reply:
Yes but we also want to enforce the child signature. So we need to pass them with the name of the argument.

Reply:
I'm not sure I understand why `super(Scaffold, self).__init__(*args, **kwargs)` doesn't work in that case. It's not changing the child class signature :thinking:

Reply:
On a side note, the new syntax recommended in Python 3 is `super()` instead of `super(MyClass, self)` https://peps.python.org/pep-3135/

Reply:
There might be something I don't understand.. could you do the suggestion of what you would like the scaffold __init__ to look like ?

Reply:
Ok so I made some checks too quickly, and it's actually not working as I want it to be. 

The two options are: 
1. Keep the child signature as is `def __init__(algo, aggregation_lr)`, and pass all arguments by hand to the parent. We could try adding a more automated unit test to check that we don't forget to pass new arguments in the future.
2. Use `def __init__(*args, **kwargs)`, and `super().__init__(*args, **kwargs)`. The check to enforce that all non-default parameters have been given will occur a few lines later than if it was in the signature, but that's probably not an issue. The largest issue in my opinion concerns readability (can be alleviated with docstrings) and auto-completion in IDE and notebooks.

Reply:
Agreed